components/mixins/profile/profile-consent.schema.json

```diff
@@ -0,0 +1,36 @@
+{
+  "meta:license": [
+    "Copyright 2020 Adobe Systems Incorporated. All rights reserved.",
+    "This work is licensed under a Creative Commons Attribution 4.0 International (CC BY 4.0) license",
+    "you may not use this file except in compliance with the License. You may obtain a copy",
+    "of the License at https://creativecommons.org/licenses/by/4.0/"
+  ],
+  "$id": "https://ns.adobe.com/xdm/mixin/profile-consent",
+  "$schema": "http://json-schema.org/draft-06/schema#",
+  "title": "Privacy/Marketing Preferences (Consent)",
+  "description": "This schema captures privacy, personalization and marketing preferences (consents).",
+  "type": "object",
+  "meta:extensible": true,
+  "meta:abstract": true,
+  "meta:intendedToExtend": ["https://ns.adobe.com/xdm/context/profile"],
+  "definitions": {
+    "profile-consents": {
```

Comment:
I see a field called "source" in "consentMetadata" which indicates we may want to store the consent information from different sources but I am not sure how we will do that with this structure.

Reply:
Each of the 36 consents has its own source field. However, for a given profile fragment (or experience event if we include consents in them) the values will generally be identical. Rather than specify the same source value (or timestamp) 36 times, simply specify it once in the metadata section and it applies to all.

Reply:
So, what you are saying is that generally each source will have its own dataset id and hence will fall in different fragments. I guess that makes sense.